src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -335,11 +410,24 @@ public void WriteEventWithData(bool SourceFlag)
             try
             {
                 EventLog.CreateEventSource(source, log);
-                EventLogEntry eventLogEntry;
-                if (SourceFlag)
-                    eventLogEntry = WriteLogEntryEventSource(source, data: true);
-                else
-                    eventLogEntry = WriteLogEntryEvent(source, data: true);
+                EventLogEntry eventLogEntry = null;
+                bool entryWritten = true;
+                while (entryWritten)
+                {
+                    try
+                    {
+                        if (SourceFlag)
+                            eventLogEntry = WriteLogEntryEventSource(source, data: true);
+                        else
+                            eventLogEntry = WriteLogEntryEvent(source, data: true);
+
+                        entryWritten = false;
+                    }
+                    catch (Win32Exception)
```

Comment:
Why not create a utility function that takes a Func<T> (in this case Func<EventLogEntry>) and invokes it with some retries in the case of a Win32Exception? Then you would not make this test any longer.

Reply:
And you would avoid all this duplicated code in this change. Duplicated code is always a bad sign.